frigate/detectors/plugins/openvino.py

```diff
@@ -17,50 +17,119 @@ class OvDetectorConfig(BaseDetectorConfig):
     type: Literal[DETECTOR_KEY]
     device: str = Field(default=None, title="Device Type")
 
-
 class OvDetector(DetectionApi):
     type_key = DETECTOR_KEY
 
     def __init__(self, detector_config: OvDetectorConfig):
         self.ov_core = ov.Core()
         self.ov_model = self.ov_core.read_model(detector_config.model.path)
+        self.ov_model_type = detector_config.model.model_type
+
+        self.h = detector_config.model.height
+        self.w = detector_config.model.width
 
         self.interpreter = self.ov_core.compile_model(
             model=self.ov_model, device_name=detector_config.device
         )
+
         logger.info(f"Model Input Shape: {self.interpreter.input(0).shape}")
         self.output_indexes = 0
+
         while True:
             try:
                 tensor_shape = self.interpreter.output(self.output_indexes).shape
                 logger.info(f"Model Output-{self.output_indexes} Shape: {tensor_shape}")
+                logger.info(f"Model Output-{self.output_indexes} Shape: {tensor_shape}")
                 self.output_indexes += 1
             except:
                 logger.info(f"Model has {self.output_indexes} Output Tensors")
                 break
 
-    def detect_raw(self, tensor_input):
+        if(self.ov_model_type == ModelTypeEnum.yolox):
+            self.num_classes = tensor_shape[2]-5
+            logger.info(f"YOLOX model has {self.num_classes} classes")
+            self.set_strides_grids()
+
+    def set_strides_grids(self):
+        grids = []
+        expanded_strides = []
+
+        strides = [8, 16, 32]
```

Comment:
Can you comment where these strides come from?

Reply:
See the original YOLOX repo [here](https://github.com/Megvii-BaseDetection/YOLOX/blob/ac379df3c97d1835ebd319afad0c031c36d03f36/yolox/utils/demo_utils.py). Also, OpenVINO models API has this too, see [here](https://github.com/openvinotoolkit/open_model_zoo/blob/e9b5cfd854d0746c1d692b77347608b47cd0c30e/demos/common/python/openvino/model_zoo/model_api/models/yolo.py).